### PR TITLE
[FIX] fix device issue for get_motion_kernel2d

### PR DESCRIPTION
**Pull request by wangguanzhi**
### Description
A few sentences describing the changes proposed in this pull request.
Fix device issue for get_motion_kernel2d.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: kornia/filters/kernels.py [hunk 1]
```diff
@@ -641,19 +641,27 @@ def get_motion_kernel2d(kernel_size: int, angle: Union[torch.Tensor, float],
...
                     [0.2195, 0.0743, 0.0000]])
     """
+
+    if isinstance(angle, torch.Tensor):
+        device = angle.device
+    elif isinstance(direction, torch.Tensor):
+        device = direction.device
+    else:
+        device = "cpu"
+
```
[the remark this answers is not in the document]
Comment (wangguanzhi) [reply kornia/filters/kernels.py]: Just pushed a new commit. Could you please take a look? Also, do you want me to move the `_extract_device_dtype` from `kornia.augmentations.utils` to `kornia.utils`?

File: kornia/filters/kernels.py [hunk 2]
```diff
@@ -641,19 +642,25 @@ def get_motion_kernel2d(kernel_size: int, angle: Union[torch.Tensor, float],
...
+        angle if isinstance(angle, torch.Tensor) else None,
+        direction if isinstance(direction, torch.Tensor) else None,
+    ])
+
     if not isinstance(kernel_size, int) or kernel_size % 2 == 0 or kernel_size < 3:
         raise TypeError("ksize must be an odd integer >= than 3")
 
     if not isinstance(angle, torch.Tensor):
-        angle = torch.tensor([angle])
+        angle = torch.tensor([angle], device=device, dtype=torch.float)
```
Comment (shijianjian) [inline kornia/filters/kernels.py:655]: Shouldn't it be as following?
```suggestion
        angle = torch.tensor([angle], device=device, dtype=dtype)
```

Comment (shijianjian) [reply kornia/filters/kernels.py:655]: Same for the rest I think

File: kornia/filters/kernels.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 from math import sqrt
 
 from kornia.geometry.transform.affwarp import rotate, rotate3d
-
+from ..augmentation.utils import _extract_device_dtype
```
Comment (shijianjian) [inline kornia/filters/kernels.py:7]: Yes. Move it to ```kornia.utils```. We should not depend on augmentation module functions.

**Review by shijianjian — CHANGES_REQUESTED** [review]

File: kornia/filters/kernels.py (hunk 2 above)
Comment (wangguanzhi) [reply kornia/filters/kernels.py:655]: Yeah. But it's also possible that `dtype` is None, right? It happens when neither `angle` nor `direction` is `torch.Tensor`.

Probably change it to
```suggestion
        angle = torch.tensor([angle], device=device, dtype=dtype if dtype else torch.float)
```

What do you think?

Comment (shijianjian) [reply kornia/filters/kernels.py:655]: If I remember right, it will be set to float32 by default when ```dtype==None```.

Comment (wangguanzhi) [reply kornia/filters/kernels.py:655]: Yes, you are right! 😁

**Comment by wangguanzhi** [comment]
Updated!

File: kornia/augmentation/utils/helpers.py [hunk 4]
```diff
@@ -3,6 +3,8 @@
 import torch
 from torch.distributions import Uniform, Beta
 
+from kornia.utils import _extract_device_dtype
```
Comment (shijianjian) [inline kornia/augmentation/utils/helpers.py:6]: Is this redundant?

**Review by shijianjian — APPROVED** [review]
LGTM. Could you check if that line is redundant or not?

File: kornia/augmentation/utils/helpers.py (hunk 4 above)
Comment (wangguanzhi) [reply kornia/augmentation/utils/helpers.py:6]: No, it's not redundant. It's used by `_adapted_uniform` and `_adapted_beta`.

File: kornia/utils/helpers.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+                raise ValueError(f"Expected None or Tensor. Got {tensor}.")
+            _device = tensor.device
+            _dtype = tensor.dtype
+            if device is None and dtype is None:
+                device = _device
+                dtype = _dtype
+            elif device != _device or dtype != _dtype:
+                raise ValueError("Passed values are not in the same device and dtype."
+                                 f"Got ({device}, {dtype}) and ({_device}, {_dtype}).")
+    return (device, dtype)
```
Comment (edgarriba) [inline kornia/utils/helpers.py:24]: @shijianjian this should have  minimal unit test

Comment (edgarriba) [reply kornia/utils/helpers.py:24]: couple of comments, no typing in the output and input should be `List[torch.Tensor] -> Tuple[torch.device, torch.dtype]`
